File: node/challenge2.js
// Implement map

function map(xs, f) {
    const result = []
    for (let i = 0; i < xs.length; i++) {
        result.push(f(xs[i]))
    }
    return result
}
const lst = [1, 2, 3, 4, 5]
const res = map(lst, (x) => x * 2)
console.log(res)
  

File: python/challenge2.py
#Implement reduceRight using reduce and reversed.

from typing import List
from functools import reduce

def reduce_right(xs: List, z, f):
    return reduce(lambda acc, x: f(x, acc), reversed(xs), z)

lst = [1, 2, 3]
res = reduce_right(lst, 0, lambda x, y: x + y)
print(res)
